### DISCUSSION TITLE
Automated Testing

### DISCUSSION THREAD

**Discussion by SE2Dev** [biodiv/contactjs#21] [Ideas]
The current testing model functions a bit more like a set of examples, and largely relies on human input to verify that things are functioning correctly. We may want to implement some level of automated testing using Jest or something similar, we may be able to emulate user input events programatically to create reproducible tests.

We should probably convert the tests to TypeScript at some point so ensure that they're up-to-date with the latest API at some point too.

**Comment by biodiv** [comment]
I agree, we need automated tests. I will start writing unit tests soon. As this will take some time, we should think about releasing v2.0, even if there are no automated unit tests yet.
The manual tests should be kept to test if things "feel right", eg in terms of thresholds,  if a real touch input is made.

**Reply by SE2Dev** [reply]
Agreed
